packages/query-core/src/hydration.ts

```diff
@@ -178,9 +183,17 @@ export function hydrate(
     }
 
     if (promise) {
+      const transformPromise =
+        client.getDefaultOptions().hydrate?.transformPromise ??
+        defaultTransformPromise
+
+      // Note: `Promise.resolve` required cause
+      // RSC transformed promises are not thenable
+      const initialPromise = transformPromise(Promise.resolve(promise))
+
```

Comment:
as discussed in Discord, let's change this to:

```
const transformPromise = client.getDefaultOptions.hydrate?.transformPromise

// Note: `Promise.resolve` required cause
// RSC transformed promises are not thenable
const initialPromise = transformPromise?.(Promise.resolve(promise)) ?? promise
```

That also means we don't need `defaultTransformPromise`